db/hints/sync_point.cc

```diff
@@ -135,13 +155,17 @@ sstring sync_point::encode() const {
     seastar::measuring_output_stream measure;
     ser::serializer<sync_point_v1>::write(measure, v1);
 
-    // Reserve version_size bytes for the version
-    sstring serialized{sstring::initialized_later{}, version_size + measure.size()};
-    seastar::simple_memory_output_stream out{serialized.data(), version_size + measure.size()};
-    ser::serializer<uint8_t>::write(out, 1);
+    // Reserve version_size bytes for the version and checksum_size bytes for the checksum
+    sstring serialized{sstring::initialized_later{}, version_size + measure.size() + checksum_size};
+
+    // Encode using V2 format
+    seastar::simple_memory_output_stream out{serialized.data(), version_size + measure.size() + checksum_size};
+    ser::serializer<uint8_t>::write(out, 2);
     ser::serializer<sync_point_v1>::write(out, v1);
+    uint64_t checksum = calculate_checksum(serialized.substr(0, version_size + measure.size()));
+    ser::serializer<uint64_t>::write(out, checksum);
 
-    return base64_encode(reinterpret_cast<int8_t*>(serialized.data(), serialized.size()));
+    return base64_encode(bytes(reinterpret_cast<signed char*>(serialized.data()), serialized.size()));
```

Comment:
`bytes_view` instead of `bytes` should be sufficient here and will avoid a copy.

Also, why change `int8_t` to `signed char`?

Reply:
For some reason I thought that `bytes` are defined with `signed char`. Fixed.